auth/jwt.go

```diff
@@ -0,0 +1,90 @@
+package auth
+
+import (
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"github.com/dgrijalva/jwt-go"
+)
+
+type JwtTokenVerifier interface {
+	VerifyToken(token string) (map[string]interface{}, error)
+	GetPrincipalWithToken(tokenString string, principal AuthenticatePrincipal) (AuthenticatePrincipal, error)
+}
+
+type AuthenticatePrincipal interface {
+	MapData(map[string]interface{}) error
+}
+
+type JwtTokenVerify struct {
```

Comment:
A struct name should be a subject. Verify is a verb, which makes this name confusing. I suggest you rename the interface: TokenVerifier and this struct to RsaTokenVerifier.

Reply:
fixed